backend/infrahub/core/query/__init__.py

```diff
@@ -648,9 +648,9 @@ def get_results_group_by(self, *args: Any) -> Generator[QueryResult, None, None]
             attrs_info[tuple(identifier)].append(info)
 
         for values in attrs_info.values():
-            attr_info = sorted(values, key=lambda i: (i["branch_score"], i["time_score"], i["deleted"]), reverse=True)[
-                0
-            ]
+            attr_info = sorted(
+                values, key=lambda i: (i["branch_score"], i["time_score"], not i["deleted"]), reverse=True
+            )[0]
```

Comment:
update the sorting order here to prefer active nodes instead of  deleted ones as a tiebreaker if the branch level and times are equal
pretty sure this is what we meant to do when this was written, but never had a situation where two IS_VISIBLE/IS_PROTECTED relationships could be updated at the same time for the same node. but the new merge query sets all timestamps at the same time, so this can happen during a merge

Reply:
We don't need to change this just now, I just wanted to highlight that we have a ruff rule (ignored) that complains about these kind of things [FURB192](https://docs.astral.sh/ruff/rules/sorted-min-max/).